src/hal/SensorInterface.h

```diff
@@ -1,9 +1,14 @@
 #ifndef SENSOR_H
 #define SENSOR_H
 
+template <typename SensorReturnType, typename SensorInputType>
+
 class SensorInterface {
 public:
-    virtual double GetAngle() = 0;
+    virtual SensorReturnType GetValue(SensorInputType) = 0;
+
+protected:
```

Comment:
Just out of curiosity, is there a specific reason to protect instead of making it private?

Reply:
A blunder from my part. I remembered incorrectly the use/properties of protected, but I double checked and there's no need for it.